docs/k8s/crds/agentendpoint.mdx

```diff
@@ -387,3 +387,185 @@ spec:
   - name: my-client-cert
     namespace: default
 ```
+
+## Status
+
+The `status` field contains the current state of the `AgentEndpoint` as reported by the ngrok operator.
+
+**Type:** `Object`
+
+**Fields:**
+
+| Field Name                                     | Type       | Description                                                                                                                           |
+| ---------------------------------------------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------------- |
+| [`status.assignedURL`](#statusassignedurl)     | `string`   | The assigned URL. This will either be the user-supplied url, or the generated assigned url depending on the configuration of spec.url |
+| 🚧 [`status.domain`](#statusdomain)            | `Object`   | TODO: Reference to the Domain resource associated with this endpoint. For internal endpoints, this will be nil                        |
+| 🚧 [`status.conditions`](#statusconditions)    | `[]Object` | Conditions describe the current conditions of the AgentEndpoint                                                                       |
+| [`status.trafficPolicy`](#statustrafficpolicy) | `string`   | Identifies any traffic policies attached to the AgentEndpoint ("inline", "none", or reference name)                                   |
```

Comment:
Suggestion from @nijikokun : potentially make this an `any` type so it can be `inline` or empty but also lets it be a ref object with name + namespace